ReflectometryServer/ChannelAccess/reflectometry_driver.py

```diff
@@ -246,3 +245,28 @@ def add_footprint_param_listeners(self):
         for parameter in parameters_to_monitor:
             parameter.add_rbv_change_listener(partial(self._update_footprint, FootprintSort.RBV))
             parameter.add_sp_rbv_change_listener(partial(self._update_footprint, FootprintSort.SP_RBV))
+
+    def _add_trigger_on_engineering_correction_change(self):
+        """
+        Add all the triggers on engineering corrections.
+
+        """
+        def _corrections_pv(pv_name, correction_update):
+            """
+            Update the driver engineering corrections PV with new value
+            Args:
+                pv_name: name fo the pv to update
+                correction_update (CorrectionUpdate): the updated values
+            Returns:
+            """
+            self._update_param_both_pv_and_pv_val(pv_name,
+                                                  correction_update.correction)
+            self.setParam("{}.DESC".format(pv_name), correction_update.description)
+            self.updatePVs()
+
+        for driver, pv_name in self._pv_manager.drivers_pv.items():
+            driver.add_listener(CorrectionUpdate, partial(_corrections_pv, pv_name))
+            last_val = driver.listener_last_value(CorrectionUpdate)
+            if last_val is None:
+                last_val = CorrectionUpdate(float("NaN"), driver.correction_description)
```

Comment:
No because we only know what to do here. We may want to throw an error or set a default, or get the value from somewhere else. I thought about adding a default value to the listener but it doesn't work because you may not have access or know what the value is.